specification/cognitiveservices/data-plane/AnomalyDetector/stable/v1.1/cadl/univariate/models.cadl

```diff
@@ -0,0 +1,304 @@
+import "@cadl-lang/rest";
+import "@azure-tools/cadl-azure-core";
+
+using Cadl.Http;
+using Cadl.Rest;
+using Azure.Core;
+
+namespace AnomalyDetector.Univariate;
+
+enum ImputeModeKnownValues {
+  Auto: "auto",
+  Previous: "previous",
+  Linear: "linear",
+  Fixed: "fixed",
+  Zero: "zero",
+  NotFill: "notFill",
+}
+
+@knownValues(ImputeModeKnownValues)
+model ImputeMode is string;
+
+enum AnomalyDetectorErrorCodesKnownValues {
+  "InvalidCustomInterval",
+  "BadArgument",
+  "InvalidGranularity",
+  "InvalidPeriod",
+  "InvalidModelArgument",
+  "InvalidSeries",
+  "InvalidJsonFormat",
+  "RequiredGranularity",
+  "RequiredSeries",
+  "InvalidImputeMode",
+  "InvalidImputeFixedValue",
+}
+
+@knownValues(AnomalyDetectorErrorCodesKnownValues)
+model AnomalyDetectorErrorCodes is string;
+
+enum TimeGranularity {
+  Yearly: "yearly",
+  Monthly: "monthly",
+  Weekly: "weekly",
+  Daily: "daily",
+  Hourly: "hourly",
+  PerMinute: "minutely",
+  PerSecond: "secondly",
+  Microsecond: "microsecond",
+  None: "none",
+}
+
+@doc("The request of entire or last anomaly detection.")
+model UnivariateDetectionOptions {
+  @doc("""
+Time series data points. Points should be sorted by timestamp in ascending
+order to match the anomaly detection result. If the data is not sorted
+correctly or there is duplicated timestamp, the API will not work. In such
+case, an error message will be returned.
+""")
+  series: TimeSeriesPoint[];
+
+  @doc("""
+Optional argument, can be one of yearly, monthly, weekly, daily, hourly,
+minutely, secondly, microsecond or none. If granularity is not present, it will
+be none by default. If granularity is none, the timestamp property in time
+series point can be absent.
+""")
+  granularity?: TimeGranularity;
+
+  @doc("""
+Custom Interval is used to set non-standard time interval, for example, if the
+series is 5 minutes, request can be set as {\"granularity\":\"minutely\",
+\"customInterval\":5}.
+""")
+  customInterval?: int32;
+
+  @doc("""
+Optional argument, periodic value of a time series. If the value is null or
+does not present, the API will determine the period automatically.
+""")
+  period?: int32;
```

Comment:
Java apiview: why period is int? What unit?

Reply:
@weidongxu-microsoft The unit of this period is decided by the granualrity, e.g. if you set your granularity as minutely, and period is 7, then this means your data has a periodical patter every 7 minutes.